### PR TITLE
neovim-nightly: fix termux_pkg_auto_update

### PR DESCRIPTION
**Pull request by licy183**
Fixes https://github.com/termux/termux-packages/pull/11290#commitcomment-79308560.

### CODE REVIEW

**Comment by truboxl** [comment]
I think this should be fixed upstream

**Comment by licy183** [comment]
The logic of generating `neovim-nightly`'s version number changes since https://github.com/neovim/neovim/commit/912dbbdd77f382d90e4d8ef8ffa483037248b83a.

In the past, it uses a lua script to get the version number. The script executes `git describe --first-parent --dirty`, gets a string like `v0.7.0-689-g6e9a7e7db`, then submatches this string to get something like `-689-g6e9a7e7db`, and concats it with `v0.8.0-dev-nightly`. Finally, it produces a string like `v0.8.0-dev-nightly-689-g6e9a7e7db`. Because `git describe` actually returns the depth between `the most recent annotated tag` and `current commit`, the version number increases as the commits are committed.

Now, it uses `git describe --first-parent --tags --always --dirty` to get the version string. It returns the depth between `the most recent tag found in refs/tags namespace` (which will always be last `nightly`) and `current commit` and gets something like `nightly-2-g6e9a7e7db`. Obviously, it doesn't increase as the commits are committed.

Currently, we are using a hack, which is parsing the release log, to get the version number of the `neovim-nightly`. We could submit such a fix to the upstream later, but I think we could just use the build timestamp, just like this PR does, to ignore the difference from upstream.

**Comment by truboxl** [comment]
If we want to use dates and time, it should be based on the upstream releases.

It should not be calculated during the bot checking for updates, which already nullifies the checking as it always produces the same version as an update.

I still feel strongly to report this to upstream as it is them who broke downstream (not just us but https://github.com/ScoopInstaller/Versions/commits/c7e95134d15b73b1886556d43f4b272a7ae62500?after=c7e95134d15b73b1886556d43f4b272a7ae62500+0&branch=c7e95134d15b73b1886556d43f4b272a7ae62500&path%5B%5D=bucket&path%5B%5D=neovim-nightly.json too). Getting from release log should not count as a hack if it worked for that long.

File: packages/neovim-nightly/build.sh [hunk 1]
```diff
@@ -63,9 +63,23 @@ termux_pkg_auto_update() {
...
+	local _current_nvim_version="$_current_nvim_version_prefix$_current_nvim_version_suffix"
+	local _remote_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$remote_nvim_version")"
+	local _remote_nvim_version_suffix="$(grep -oP '\-\d+-g[0-9a-f]+' <<< "$remote_nvim_version")"
+	local _current_time=$(date "+%Y%m%d%H%M%S")
+	local _package_version_to_update="$_remote_nvim_version_prefix$_current_time$_remote_nvim_version_suffix"
+
 	# since we are using a nightly build, therefore no need to check for version increment/decrement.
-	if [ "${TERMUX_PKG_VERSION}" != "${remote_nvim_version}" ]; then
-		termux_pkg_upgrade_version "${remote_nvim_version}" --skip-version-check
+	if [ "${_current_nvim_version}" != "${remote_nvim_version}" ]; then
```
Comment (truboxl) [inline packages/neovim-nightly/build.sh:81]: This will always be true each time the bot check for updates

Comment (licy183) [reply packages/neovim-nightly/build.sh:81]: It's not like what you think.

Let us assume that `TERMUX_PKG_VERSION="0.8.0-dev-nightly-202207250000-4-ge9b58a619"`, `remote_nvim_version=0.8.0-dev-nightly-9-g1ef84547a`.
```
_current_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$TERMUX_PKG_VERSION")" # "0.8.0-dev-nightly-"
_current_nvim_version_suffix="$(grep -oP '\d+-g[0-9a-f]+' <<< "$TERMUX_PKG_VERSION")"               # "4-ge9b58a619"
_current_nvim_version="$_current_nvim_version_prefix$_current_nvim_version_suffix"                  # "0.8.0-dev-nightly-4-ge9b58a619"
_remote_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$remote_nvim_version")" # "0.8.0-dev-nightly-"
_remote_nvim_version_suffix="$(grep -oP '\-\d+-g[0-9a-f]+' <<< "$remote_nvim_version")"             # "-9-g1ef84547a"
_current_time=$(date "+%Y%m%d%H%M%S")                                                               # "20220725225350"
_package_version_to_update="$_remote_nvim_version_prefix$_current_time$_remote_nvim_version_suffix" # "0.8.0-dev-nightly-20220725225350-9-g1ef84547a"
[ "${_current_nvim_version}" != "${remote_nvim_version}" ] # [ "0.8.0-dev-nightly-4-ge9b58a619" != "0.8.0-dev-nightly-9-g1ef84547a" ] # true
```

And if `TERMUX_PKG_VERSION="0.8.0-dev-nightly-202207250000-4-ge9b58a619"`,`remote_nvim_version=0.8.0-dev-nightly-4-ge9b58a619`,
```
_current_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$TERMUX_PKG_VERSION")" # "0.8.0-dev-nightly-"
_current_nvim_version_suffix="$(grep -oP '\d+-g[0-9a-f]+' <<< "$TERMUX_PKG_VERSION")"               # "4-ge9b58a619"
_current_nvim_version="$_current_nvim_version_prefix$_current_nvim_version_suffix"                  # "0.8.0-dev-nightly-4-ge9b58a619"
_remote_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$remote_nvim_version")" # "0.8.0-dev-nightly-"
_remote_nvim_version_suffix="$(grep -oP '\-\d+-g[0-9a-f]+' <<< "$remote_nvim_version")"             # "-4-ge9b58a619"
_current_time=$(date "+%Y%m%d%H%M%S")                                                               # "20220725225350"
_package_version_to_update="$_remote_nvim_version_prefix$_current_time$_remote_nvim_version_suffix" # "0.8.0-dev-nightly-20220725225632-4-ge9b58a619"
[ "${_current_nvim_version}" != "${remote_nvim_version}" ] # [ "0.8.0-dev-nightly-4-ge9b58a619" != "0.8.0-dev-nightly-4-ge9b58a619" ] # false
```

Why `[ "${_current_nvim_version}" != "${remote_nvim_version}" ]` will always be true?

**Comment by licy183** [comment]
> If we want to use dates and time, it should be based on the upstream releases.
>
> Timestamp (not timestrap) should not be calculated during the bot checking for updates, which already nullifies the checking as it always produces the same version as an update.

Yeah, this makes some sense. But the action bot starts about every 6 hour, and we have checked the version number is the same or not, so I think there is actually no need to use the time based on the upstream releases.

**Comment by licy183** [comment]
As upstream has fixed the version format, I think all we need to do is bumpping  epoch of the package version.

**Review by truboxl — APPROVED** [review]
I hate bumping epoch but there's no other choice I guess

Thanks for reporting to upstream!

**Pull request merged**